frontends/concrete-python/concrete/fhe/mlir/context.py

```diff
@@ -3455,6 +3458,190 @@ def reinterpret(self, x: Conversion, *, bit_width: int) -> Conversion:
         )
         return self.operation(operation, resulting_type, x.result)
 
+    def where(
+        self,
+        resulting_type: ConversionType,
+        condition: Conversion,
+        when_true: Conversion,
+        when_false: Conversion,
+    ) -> Conversion:
+        if condition.original_bit_width != 1 or condition.is_signed:
+            highlights = {
+                condition.origin: "condition is not uint1",
```

Comment:
is original_bit_width the final type ?
"condition is not 0 or 1" ?

Reply:
It's what will be displayed on the error. It's the traced and measured dtype.

Reply:
I'd prefer to keep it like this, it's very explicit in the errors.

Reply:
what I mean is I can use a final uint2 and it will work, so the message could be more precise.
I.e. I am assuming this is not rejected:
```python
@fhe....
def c(bool_a, bool_b):
   d = bool_a + bool_b # implies uint2 for a and b
   e = n.where(bool_a, [1, 2]) 
   ...
 ```

Reply:
No you can't, that's the point of the error. We pack the condition in an additional bit along with the chunk, how would we pack 2 bits or more?